src/field_theory/complete_ordered.lean

```diff
@@ -0,0 +1,769 @@
+
+This file shows that the reals are unique, or, more formally, given a type satisfying the common
+axioms of the reals (field, conditionally complete, linearly ordered) that there is an equivalence
+preserving these properties to the reals. This is `cut_ordered_ring_equiv`. Moreover this
+equivalence is unique.
+
+We introduce definitions of conditionally complete linear ordered fields, show all such are
+archimedean, and define equivalences between these fields. We also construct the natural map from a
+`linear_ordered_field` to such a field.
+
+## Main definitions
+
+* `conditionally_complete_linear_ordered_field` :
+* `ordered_ring_equiv`      :
+* `cut_map`      :
+
+## Main results
+
+* `ring_equiv_eq_cut_ordered_ring_equiv`:
+
+## Notation
+
+* `≃+*o`: equivalence of ordered rings.
+
+## References
+
+* https://mathoverflow.net/questions/362991/who-first-characterized-the-real-numbers-as-the-unique-complete-ordered-field
+
+## Tags
+reals, conditionally complete, ordered field
+-/
+
+noncomputable theory
+open_locale classical
+
+open set
+
+/-- The natural equiv between the rationals and the rationals as a set inside any characteristic 0
+    division ring -/
+def rat_equiv_of_char_zero (F : Type*) [division_ring F] [char_zero F] : ℚ ≃ range (coe : ℚ → F) :=
+equiv.set.range coe rat.cast_injective
+
+/-- The natural equiv between the rationals as a set inside any pair of characteristic 0
+    division rings -/
+def range_rat_equiv (F K : Type*) [division_ring F] [char_zero F] [division_ring K] [char_zero K] :
+  range (coe : ℚ → F) ≃ range (coe : ℚ → K) :=
+(rat_equiv_of_char_zero F).symm.trans (rat_equiv_of_char_zero K)
+
+@[simp]
+lemma range_rat_equiv_apply {F K : Type*} {q : ℚ}
+  [division_ring F] [char_zero F] [division_ring K] [char_zero K] :
+  range_rat_equiv F K ⟨q, mem_range_self q⟩ = ⟨q, mem_range_self q⟩ :=
+by simp only [range_rat_equiv, rat_equiv_of_char_zero, equiv.set.range_apply,
+  function.comp_app, subtype.mk_eq_mk, equiv.coe_trans, rat.cast_inj, equiv.symm_apply_eq]
+
+/-- The function sending subsets of the rationals embedded inside of one characteristic zero
+    division ring to the corresponding subset of a second such ring. -/
+def cut_map (F K : Type*) [division_ring F] [char_zero F] [division_ring K] [char_zero K] :
+  set (set.range (coe : ℚ → F)) → set (set.range (coe : ℚ → K)) :=
+λ c, (range_rat_equiv F K) '' c
+
+/- No longer used -/
+lemma bdd_above_add {F : Type*} [ordered_add_comm_monoid F] {A B : set F} :
+  bdd_above A → bdd_above B → bdd_above (A + B) :=
+begin
+  rintros ⟨bA, hbA⟩ ⟨bB, hbB⟩,
+  use bA + bB,
+  rintros x ⟨xa, xb, hxa, hxb, rfl⟩,
+  exact add_le_add (hbA hxa) (hbB hxb),
+end
+
+
+/-- Equivalence commuting with multiplicative, additive and order structure. -/
+@[nolint has_inhabited_instance]
+structure ordered_ring_equiv (R S : Type*) [has_mul R] [has_add R] [has_mul S] [has_add S]
+  [has_le R] [has_le S] extends R ≃+* S :=
+(map_rel_iff' : ∀ {x y : R}, to_fun x ≤ to_fun y ↔ x ≤ y)
```

Comment:
This removes the need for `to_order_iso` below
```suggestion
  [has_le R] [has_le S] extends R ≃+* S, R ≃o S
```

Reply:
Unfortunately order_iso is an abbreviation not a structure so your suggestion doesn't work.
As for enabling old structure command and using `((\le) \to R \to R \to Prop) \simeqr ((\le) \to S \to S \to Prop)` instead, I tried it that way too and I couldn't prove some things (like `ext` iirc) using that definition so I figured this way was simplest